grid_map_pcl/src/GridMapPclLoader.cpp

```diff
@@ -129,31 +129,50 @@ void GridMapPclLoader::processGridMapCell(const unsigned int linearGridMapIndex,
     return;
   }
   auto& clusterHeights = clusterHeightsWithingGridMapCell_[index(0)][index(1)];
-  calculateElevationFromPointsInsideGridMapCell(pointsInsideCellBorder, clusterHeights);
-  if (clusterHeights.empty()) {
-    (*gridMapData)(index(0), index(1)) = std::nan("1");
-  } else {
-    (*gridMapData)(index(0), index(1)) = params_.get().clusterExtraction_.useMaxHeightAsCellElevation_
-                                             ? *(std::max_element(clusterHeights.begin(), clusterHeights.end()))
-                                             : *(std::min_element(clusterHeights.begin(), clusterHeights.end()));
-  }
+  (*gridMapData)(index(0), index(1)) = calculateElevationFromPointsInsideGridMapCell(pointsInsideCellBorder, clusterHeights);
 }
 
-void GridMapPclLoader::calculateElevationFromPointsInsideGridMapCell(Pointcloud::ConstPtr cloud, std::vector<float>& heights) const {
```

Comment:
I would actually change the interface to explicitly add the height type to reduce side effects.

Reply:
This comment is not for me and you will change it?

Reply:
No sorry, if you could change it in this PR, that would be great.

Reply:
Sorry, you mean adding height_type to arguments like below?
```
float GridMapPclLoader::calculateElevationFromPointsInsideGridMapCell(Pointcloud::ConstPtr cloud, std::vector<float>& heights, const int height_type) const {
```

Reply:
Yes, that would reduce the side effects of the function.

Reply:
Added `height_type` as a arg of `calculateElevationFromPointsInsideGridMapCell`
https://github.com/ANYbotics/grid_map/pull/329/commits/98550539024f54ab72a3fff9962bb82430b57dc9